[Sonoma wk2] fast/replaced/pdf-as-embed-with-no-mime-type-is-not-blank.html is a flaky failure

aprotyas opened the pull request:
#### 932af5b1166d073e4d51829adff52c9db7ab8fbd
<pre>
[Sonoma wk2] fast/replaced/pdf-as-embed-with-no-mime-type-is-not-blank.html is a flaky failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=278429">https://bugs.webkit.org/show_bug.cgi?id=278429</a>
<a href="https://rdar.apple.com/134377511">rdar://134377511</a>

Reviewed by Wenson Hsieh.

PDFKit performs an animated fade-in of the PDF document to render, and
as such it looks like the &lt;embed&gt; just never loads. We work around this
issue by rAF-ing after a timeout of 250ms, which we&apos;ve previously
determined to be enough time for the PDF fade-in animation to pass.

Instead of adjusting the single failing test, though, we introduce a new
utility UIHelper.waitForPDFFadeIn, and adopt that in other relevant
tests.

Also, we get rid of the PluginsEnabled test preference, since we removed
support for that key in 274917@main.

* LayoutTests/compositing/plugins/pdf/pdf-in-embed-rounded-border.html:
* LayoutTests/compositing/plugins/pdf/pdf-in-embed-with-root-page-zoom.html:
* LayoutTests/compositing/plugins/pdf/pdf-in-embed.html:
* LayoutTests/compositing/plugins/pdf/pdf-in-styled-embed.html:
* LayoutTests/compositing/plugins/pdf/pdf-scrolling-tree.html:
* LayoutTests/fast/replaced/pdf-as-embed-with-no-mime-type-is-not-blank.html:
* LayoutTests/http/tests/pdf/linearized-pdf-in-iframe.html:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async waitForPDFFadeIn):
(window.UIHelper):

Canonical link: <a href="https://commits.webkit.org/282638@main">https://commits.webkit.org/282638@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c84e71bdce7673664c92ae801098457bad88e2ac

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63786 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/43143 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16383 "Failed to checkout and rebase branch from PR 32554") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/67807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/14394 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/50833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/14674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/67807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/14394 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66855 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/50833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/55/builds/16383 "Failed to checkout and rebase branch from PR 32554") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/67807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/50833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/55/builds/16383 "Failed to checkout and rebase branch from PR 32554") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/13267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/50833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/55/builds/16383 "Failed to checkout and rebase branch from PR 32554") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/69503 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/87/builds/7733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/14674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/69503 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/86/builds/7766 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/55/builds/16383 "Failed to checkout and rebase branch from PR 32554") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/69503 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9643 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/38963 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/40042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/41225 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/39785 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

whsieh reviewed (approved):
Nit - tabs -> 4 spaces

aprotyas committed:
[Sonoma wk2] fast/replaced/pdf-as-embed-with-no-mime-type-is-not-blank.html is a flaky failure
https://bugs.webkit.org/show_bug.cgi?id=278429
rdar://134377511

Reviewed by Wenson Hsieh.

PDFKit performs an animated fade-in of the PDF document to render, and
as such it looks like the <embed> just never loads. We work around this
issue by rAF-ing after a timeout of 250ms, which we've previously
determined to be enough time for the PDF fade-in animation to pass.

Instead of adjusting the single failing test, though, we introduce a new
utility UIHelper.waitForPDFFadeIn, and adopt that in other relevant
tests.

Also, we get rid of the PluginsEnabled test preference, since we removed
support for that key in 274917@main.

* LayoutTests/compositing/plugins/pdf/pdf-in-embed-rounded-border.html:
* LayoutTests/compositing/plugins/pdf/pdf-in-embed-with-root-page-zoom.html:
* LayoutTests/compositing/plugins/pdf/pdf-in-embed.html:
* LayoutTests/compositing/plugins/pdf/pdf-in-styled-embed.html:
* LayoutTests/compositing/plugins/pdf/pdf-scrolling-tree.html:
* LayoutTests/fast/replaced/pdf-as-embed-with-no-mime-type-is-not-blank.html:
* LayoutTests/http/tests/pdf/linearized-pdf-in-iframe.html:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/resources/ui-helper.js:
(window.UIHelper.async waitForPDFFadeIn):
(window.UIHelper):

Canonical link: https://commits.webkit.org/282638@main
